python/HeaderService/hslib_salobj.py

```diff
@@ -155,12 +155,75 @@ def get_ip(self):
         self.ip_address = socket.gethostbyname(socket.gethostname())
         self.log.info("Will use IP: {} for web service".format(self.ip_address))
 
+    def get_s3instance(self):
+        """
+        Figure the location where are running to define the s3instance
+        in case it wasn't defined.
+        """
+        # Check if defined in self.config or the environment
+        if self.config.s3instance:
+            s3instance = self.config.s3instance
+            self.log.info(f"Will use s3instance in config: {s3instance}")
+            return s3instance
+        elif 'S3INSTANCE' in os.environ:
+            s3instance = os.environ['S3INSTANCE']
+            self.log.info(f"Will use s3instance from environment: {s3instance}")
+            return s3instance
+
+        # Try to auto-figure out from location
+        self.log.warning("The s3instance was not defined in config")
+        address = socket.getfqdn()
+        if address.find('.ncsa.') >= 0:
+            s3instance = 'nts'
+        elif address.find('tuc') >= 0:
+            s3instance = 'tuc'
+        elif address.find('.cp.') >= 0:
+            s3instance = 'cp'
+        else:
+            s3instance = 'dummy'
+        self.log.info(f"Will use auto-config s3instance: {s3instance}")
```

Comment:
I like that the environment variable S3INSTANCE is used to configure this, and it might be best to require that it be set, and fail otherwise.  If in the future the domain name at the base is aliased or changes for some reason and "cp" isn't there, then it could default to "dummy", which could lead to some confusion down the road.

Reply:
I don't think we have a clear policy on how the s3instances would work yet. So far I've left it pretty flexible, as the s3instance can in addition be defined in the configuration file and on the command-line with `--s3instance`  option.